src/main.rs

```diff
@@ -184,13 +210,22 @@ fn main() -> Result<(), String> {
                     ];
                     set_shape(&conn, window, &rects);
                 } else if !opt.no_guides {
-                    let rects = build_guides(
-                        screen_rect,
-                        xcb::Point::new(motion.event_x(), motion.event_y()),
-                        guide_width,
+                    let result = get_window_at_point(
+                        &conn,
+                        root,
+                        cursor_pt,
+                        opt.remove_decorations,
+                        Some(window),
                     );
-
-                    set_shape(&conn, window, &rects);
+                    let guides = build_guides(screen_rect, cursor_pt, guide_width);
+                    if let Some(result) = result {
+                        let borders = build_border(result.rect, line_width);
+                        let mut rects = guides.to_vec();
```

Comment:
We could probably construct the new array manually:             let rects = [guides[0], guides[1], borders[0], borders[1], borders[2], borders[3]];

Reply:
Yes, I thought about this. Too bad no `const-generics` for `[arr1, arr2].concat()`.